File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.py
# Last updated: 10/22/2025, 12:09:15 AM
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        windowStart = 0
        maxLength = 0
        oneCount = 0

        for windowEnd in range(len(nums)):
            if nums[windowEnd] == 1:
                oneCount += 1

            while (windowEnd - windowStart + 1 - oneCount) > k:
                if nums[windowStart] == 1:
                    oneCount -= 1
                windowStart += 1

            maxLength = max(maxLength,windowEnd - windowStart + 1)
        return maxLength
        

File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.js
// Last updated: 10/22/2025, 12:09:13 AM
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
    let left = 0
    let ans = 0
    let currZeroes = 0
    for(let right = 0;right < nums.length;right++){
        if(nums[right] === 0){
            currZeroes+=1
        }
        while(currZeroes > k){
            if(nums[left] === 0){
                currZeroes-=1
            }
            left++
        }
        ans = Math.max(ans,right - left + 1)
    }

    return ans
    
};